stack/eslint-plugin/src/no-untyped-underscore.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import {
+  TSESTree,
+  TSESLint,
+  ParserServices
+} from '@typescript-eslint/experimental-utils';
+import * as ts from 'typescript';
+
+type MessageIds = 'error-untyped-underscore';
+type Options = [ ];
+
+const noUntypedUnderscoreRule: TSESLint.RuleModule<MessageIds,Options> = {
+  meta: {
+    type: 'problem',
+    messages: {
+      'error-untyped-underscore':
+        'This expression appears to access a private member "{{memberName}}"; '
+        + 'either remove the underscore prefix or else declare a type for the containing object'
+    },
+    schema: [ ],
+    docs: {
+      description: 'Prevent TypeScript code from accessing legacy JavaScript members'
+        + ' whose name has an underscore prefix',
+      category: 'Stylistic Issues',
+      recommended: false,
+      url: 'https://www.npmjs.com/package/@rushstack/eslint-plugin'
+    }
+  },
+  create: (context: TSESLint.RuleContext<MessageIds, Options>) => {
+
+    const parserServices: ParserServices | undefined = context.parserServices;
+    if (!parserServices ||
+      !parserServices.program ||
+      !parserServices.esTreeNodeToTSNodeMap) {
+        throw new Error('This rule requires your ESLint configuration to define the "parserOptions.project"'
```

Comment:
```suggestion
        throw new Error('This rule requires your ESLint configuration to define the "parserOptions.program"'
```

Reply:
Won't fix: The ESLint setting really is called `parserOptions.project`.  The `parserServices.program` API is a separate consequence of that.